ldap/ldap.go

```diff
@@ -104,6 +94,71 @@ func (l *Config) Connect() (ldapConn *ldap.Conn, err error) {
 	return ldapConn, err
 }
 
+// Connect connect to ldap server.
+func (l *Config) Connect() (ldapConn *ldap.Conn, err error) {
+	if l == nil || !l.Enabled {
+		return nil, errors.New("LDAP is not configured")
+	}
+
+	var srvService, srvProto, srvName string
+	switch l.SRVRecordName {
+	case "on":
+		srvName = l.ServerAddr
+	case "ldap", "ldaps":
+		srvService = l.SRVRecordName
+		srvProto = "tcp"
+		srvName = l.ServerAddr
+	case "":
+	default:
+		return nil, errors.New("Invalid SRV Record Name parameter")
+
+	}
+
+	if srvName == "" {
+		// No SRV Record lookup case.
+		ldapAddr := l.ServerAddr
+
+		_, _, err = net.SplitHostPort(ldapAddr)
+		if err != nil {
+			if strings.Contains(err.Error(), "missing port in address") {
+				// Use default LDAP port if none specified "636"
+				ldapAddr = net.JoinHostPort(ldapAddr, "636")
+			} else {
+				return nil, err
+			}
+		}
+
+		return l.connect(ldapAddr)
+	}
+
+	// SRV Record lookup is enabled.
+	_, addrs, err := net.LookupSRV(srvService, srvProto, srvName)
```

Comment:
Just wanted a confirmation, does net.LookupSRV() work if srvServce == "" and srvProto == "" ? this is when SRVRecordName == "on"

Reply:
Yes, this is the documented behavior - https://pkg.go.dev/net#LookupSRV